script/search/2.0/search.inc.php

```diff
@@ -153,10 +154,20 @@ private function GetSearchQueries(KeyboardSearchResult $result) {
         $stmt->bindParam(5, $result->pageNumber, PDO::PARAM_INT);
         $stmt->bindParam(6, $result->pageSize, PDO::PARAM_INT);
         break;
-      case KeyboardSearchResult::FILTER_POPULARITY:
-        // list most popular keyboards (skip obsolete always)
-        $result->rangetext = "Popular keyboards";
-        $stmt = $this->new_query('EXEC sp_keyboard_search_by_popularity ?, ?, ?');
+      case KeyboardSearchResult::FILTER_PREDEFINED:
+        // returns predefined search results (skip obsolete always)
+        if($result->text == '*' || substr('popularity', 0, strlen($result->text)) == $result->text) {
+          $result->rangetext = "Popular keyboards";
+          $stmt = $this->new_query('EXEC sp_keyboard_search_by_popularity ?, ?, ?');
+          $clampTotalRows = 100;
+        } else if($result->text == '!' || substr('alphabetically', 0, strlen($result->text)) == $result->text) {
```

Comment:
Compare against https://github.com/keymanapp/keyman.com/blob/6f350c478c0b8b65fe4239342398b4413636f3c0/keyboards/index.php#L47 (from https://github.com/keymanapp/keyman.com/pull/242):

```
    <p>Enter the name of a keyboard or language to search for. (<a href="?q=p:popular">Popular keyboards</a> | <a href="?q=p:alphabetical">All keyboards</a>)</p>
```

I'd like to note that `'alphabetical' != 'alphabetically'`.  That said, after looking up PHP's definition for `substr`... I guess this _does_ cover both variants.  It's kind of odd to use `alphabetical` on one site but `alphabetically` on the other, though.  Why introduce ambiguity on what's essentially an internal parameter?

Reply:
Well, the weird and wonderful reason I had was I wanted to accept as much as possible but provide the most appropriate version for use in the url -- so it will take `p:a`, `p:alpha`, `p:alphabetic`, `p:alphabetically` etc. I could have just gone with `alphabetic` but I liked the way `alphabetically` rhymed sorta with `popularity`. It is good reasons it is.

Reply:
Wait... will it?  https://github.com/keymanapp/api.keyman.com/blob/05c950d319533fc99fd402ae416f2db93e3157e0/tests/Search20Test.php#L294-L296

Reply:
I wouldn't expect `p:a` or `p:alpha` to work based on the block I just linked.

Reply:
Wait... right, that's a unit-test block.  Might help to keep that straight.